### PR TITLE
chore: add eslint and eslint-config-airbnb

### PR DESCRIPTION
**Pull request by stu01509**
add eslint and eslint-config-airbnb to devDependencies

below added by bradymiller:
Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Alpha_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by stu01509** [comment]
Hi @tywrenn  and @bradymiller 

I already import the eslint, eslint-airbnb-config, and rule file, please take look.
If haven't any problem, I will start to convert the javascript file in our current codebase(except the gulp.js).

**Comment by tywrenn** [comment]
Ran ESLint - it immediately crashed XD

**Comment by tywrenn** [comment]
Probably will need to develop a CLI command that doesn't kill everything

**Comment by stu01509** [comment]
Hey @tywrenn 

So should I convert js file in this PR?
Or waiting for this one merged then create another one?

**Comment by tywrenn** [comment]
@stu01509 You should convert/fix the JS in this PR

**Comment by bradymiller** [comment]
hi @stu01509 ,
Are all these changes automatic or manual or hybrid.
Looks like we are all gonna be taking a crash course in modern javascript development :)

**Comment by stu01509** [comment]
Hi @bradymiller 

All these changes almost by manual, because some javascript depends on other libraries or pages, so the automatic changes aren't reliable.

Also, are you any suggestion which folder's javascript needs a priority to convert to under eslint rules?

**Comment by bradymiller** [comment]
Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Alpha_-_Up_For_Grabs_Demo

File: interface/modules/zend_modules/public/js/autosuggest/autosuggest.js [hunk 1]
```diff
@@ -19,96 +19,99 @@
...
+        //                     $('#autoSuggestionsList').html(data);
+        //                 }
+        //                 // Focus to 1st Row
+        //                 if (document.getElementById('list_' + searchEleNo + '_1')) {
+        //                     $('#list_' + searchEleNo + '_1').focus();
+        //                     $('#list_' + searchEleNo + '_1').css("background-color", "#659CD8");
+        //                 }
+        //             }
+        //         }
+        //     });
```
Comment (stu01509) [inline interface/modules/zend_modules/public/js/autosuggest/autosuggest.js:112]: Hi  @bradymiller 

Not sure about this code, guess they are unnecessary or brokered,
so I commented out the code need you to confirm.

**Comment by stu01509** [comment]
Hi @bradymiller  and @tywrenn 

The javascript coding-style standardize are done, we can start to review now.

**Comment by tywrenn** [comment]
@stu01509 What command are you using to check for lint issues?

**Comment by stu01509** [comment]
> @stu01509 What command are you using to check for lint issues?

No, I didn't add yet, just thinking we can start to review that, I will add command later.

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "dev": "gulp --dev && gulp watch",
     "gulp-build": "gulp -b --dev",
     "gulp-watch": "gulp watch",
+    "lint": "eslint **/*.js",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
```
Comment (stu01509) [inline package.json:15]: Hi @tywrenn  and @bradymiller 

Finally, I add the `npm run lint` to lint all javascript files, please take a look.

Comment (bradymiller) [reply package.json:15]: neat, then we can run this from travis when this is in the codebase

**Comment by bradymiller** [comment]
hi @stu01509 ,
Just did some quick testing on the up for grab demo for this PR at https://www.open-emr.org/wiki/index.php/Development_Demo#Alpha_-_Up_For_Grabs_Demo and noticed quite a bit of javascript errors.

File: package.json (hunk 2 above)
Comment (stu01509) [reply package.json:15]: Yeah, should I follow the .travis.yml file to add my command?

**Comment by stu01509** [comment]
Hi @bradymiller 

Fix the product registration js error.
Do you have any idea of testing all javascript?

**Comment by bradymiller** [comment]
hi @stu01509 , Think were stuck doing old fashioned manual testing unless @tywrenn has better idea :)

**Comment by tywrenn** [comment]
I've had a lot going on -- will check this when I have time

**Comment by stu01509** [comment]
Hi @tywrenn 

Should I resolve the conflicting files?

**Comment by tywrenn** [comment]
@stu01509 yeah go ahead and do that -- I'm trying to find spare time at the moment 😂😂

**Comment by bradymiller** [comment]
bump

**Pull request closed without merging**